src/commands.rs

```diff
@@ -4022,78 +4050,138 @@ fn cmd_branch(ui: &mut Ui, command: &CommandHelper, args: &BranchArgs) -> Result
                     branch_name
                 ))),
             })
-            .try_collect()?
-    } else {
-        args.names.iter().map(|name| name.as_str()).collect()
-    };
+            .try_collect()?;
+        Ok(result)
+    }
 
-    if branch_names.is_empty() {
-        ui.write_warn("warning: No branches provided.\n")?;
+    fn make_branch_term(ui: &mut Ui, branch_names: &[impl AsRef<str>]) -> io::Result<String> {
+        if branch_names.is_empty() {
+            ui.write_warn("warning: No branches provided.\n")?;
+        }
+        let branch_term = if branch_names.len() == 1 {
+            "branch"
+        } else {
+            "branches"
+        };
+        Ok(format!(
+            "{branch_term} {}",
+            branch_names
+                .iter()
+                .map(|branch_name| branch_name.as_ref())
+                .join(", ")
+        ))
```

Comment:
Maybe it's clearer to special-case the single-branch case with `Ok(format!("branch {}", branch_names[0].as_ref()))` (and removing the `branch_term`)?